### PR TITLE
Fix pose data format

### PR DESCRIPTION
**Pull request by Gamenot**
- Force `Pose.position` to be a length 3 `np.array`.
- Force `Pose.orientation` to be a length 4 `np.array`.
- Fix various type issues

See #1275

### CODE REVIEW

File: smarts/core/coordinates.py [hunk 1]
```diff
@@ -258,12 +256,20 @@ class Pose:
...
-    orientation: Union[
-        NamedTuple, Tuple, np.ndarray
-    ]  # [a, b, c, d] -> a + bi + cj + dk = 0
+    position: np.ndarray  # [x, y, z]
+    orientation: np.ndarray  # [a, b, c, d] -> a + bi + cj + dk = 0
     heading_: Optional[Heading] = None  # cached heading to avoid recomputing
 
+    def __post_init__(self):
+        if not isinstance(self.position, np.ndarray):
+            self.position = np.array(self.position, dtype=np.float64)
```
Comment (RutvikGupta) [inline smarts/core/coordinates.py:265]: You should use `dtype=float` since `np.float64` is deprecated in version of numpy.

File: smarts/core/utils/math.py [hunk 2]
```diff
@@ -307,9 +307,6 @@ def position_at_offset(
     if is_close(dist, offset):
         return p2
 
-    if offset > dist:
-        return None
-
     return p1[0] + (p2[0] - p1[0]) * (offset / dist), p1[1] + (p2[1] - p1[1]) * (
```
Comment (RutvikGupta) [inline smarts/core/utils/math.py:310]: Why was this removed?

File: smarts/core/tire_models.py [hunk 3]
```diff
@@ -116,6 +116,10 @@ def _calculate_slip_angles(chassis, client, action):
 
         return slip_angles
 
+    def apply_tire_forces(self, chassis, client, action):
+        """Applies tire forces"""
+        raise NotImplementedError
```
Comment (RutvikGupta) [inline smarts/core/tire_models.py:121]: Will this throw an error since it is being used in `smarts/core/chassis.py` but not implemented?

**Review by RutvikGupta — APPROVED** [review]

File: smarts/core/coordinates.py [hunk 4]
```diff
@@ -258,12 +256,20 @@ class Pose:
     """A pair of position and orientation values."""
 
     # TODO: these should be np.ndarray
```
Comment (Adaickalavan) [inline smarts/core/coordinates.py:258]: Can this `# TODO` be removed now, since change to np.ndarray has been made?

File: smarts/core/sumo_road_network.py [hunk 5]
```diff
@@ -150,15 +150,19 @@ def from_spec(cls, map_spec: MapSpec):
         # Connections to internal lanes are implicit. If `withInternal=True` is
         # set internal junctions and the connections from internal lanes are
         # loaded into the network graph.
+        # pytype: disable=module-attr
         G = sumolib.net.readNet(net_file, withInternal=True)
+        # pytype: enable=module-attr
```
Comment (Adaickalavan) [inline smarts/core/sumo_road_network.py:155]:
```suggestion
        G = sumolib.net.readNet(net_file, withInternal=True)  # pytype: disable=module-attr
```
If disabling pytype for a single line, perhaps use a single line pytype disable comment.

File: smarts/core/sumo_road_network.py [hunk 6]
```diff
@@ -150,15 +150,19 @@ def from_spec(cls, map_spec: MapSpec):
...
 
         if not cls._check_net_origin(G.getBoundary()):
             shifted_net_file = cls.shifted_net_file_path(net_file)
             if os.path.isfile(shifted_net_file) or (
                 map_spec.shift_to_origin
                 and cls._shift_coordinates(net_file, shifted_net_file)
             ):
+                # pytype: disable=module-attr
                 G = sumolib.net.readNet(shifted_net_file, withInternal=True)
+                # pytype: enable=module-attr
```
Comment (Adaickalavan) [inline smarts/core/sumo_road_network.py:165]: Similar suggestion, as previous, to use a single line pytype disable comment.
```suggestion
                G = sumolib.net.readNet(shifted_net_file, withInternal=True)  # pytype: disable=module-attr
```

**Review by Adaickalavan — APPROVED** [review]

File: smarts/core/sumo_road_network.py (hunk 5 above)
Comment (Gamenot) [reply smarts/core/sumo_road_network.py:155]: I tried to do that but the `black` formatting caused issues.

File: smarts/core/utils/math.py (hunk 2 above)
Comment (Gamenot) [reply smarts/core/utils/math.py:312]: This case never gets hit, I looked through the code and found that if this case would get hit it would cause errors in the code since we always unpack the result without checking checking the truthy-ness.

File: smarts/core/coordinates.py (hunk 1 above)
Comment (Gamenot) [reply smarts/core/coordinates.py:265]: `np.float64` does not seem to be deprecated although `np.float` is.
